### PR TITLE
Added Deque Directory

### LINKED ISSUE #621: Doubly Ended Queue - Deque Implementation
### Description

I've gone through the codebase & found out that there is no directory with the implementation of the deque. I would like to add the basic operations of deque, including push & pop from start & end.
Kindly assign me this issue.

### Domain

Data Structure

### Type of Contribution

Addition

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) & [Code of conduct](https://github.com/Lakhankumawat/LearnCPP/blob/main/.github/CODE_OF_CONDUCT.md) of LearnCPP.

### PR DESCRIPTION
**Pull request by eshaan5**
## Related Issue
- This PR closes #621 .

## Proposed Changes
- Added a basic implementation of Deque in C++.
- Changed the README of Queue accordingly.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: D-Deque/slidingWindowMaximum.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#include <iostream>
```
Comment (Lakhankumawat) [inline D-Deque/slidingWindowMaximum.cpp:1]: Delete this file we only allow one code file with README at a time

File: D-Deque/DequeSTL.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#include <iostream>
```
Comment (Lakhankumawat) [inline D-Deque/DequeSTL.cpp:1]: Change Your code directory your code must be present inside Q-Queue , Deque is a part of Queue and you need to add your changes inside Q-Queue's README.md

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

**Comment by Lakhankumawat** [comment]
Make sure your Pull Request is descriptive and complete the checklist .
I would recommend you to go through this tutorial video of the project [here](https://youtu.be/kIFwqp9B7pQ) ,thanks.


///------------------Description -----------------------///

For contribution and more guidelines i would recommend you to go through this 
tutorial video of the project [here](https://youtu.be/7UgC3HDETRA), thanks.



///------------------Description -----------------------///

Let me show you how to check the check list , there should be no space inside brackets like [x]
 `[x]`
- [x] Checklist
- [ ] Or you can just check them from outside without doing any of these
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [ ] 🌟 ed the repo

1. Starr the repo if you haven't !
2. Have you watched this [video](https://youtu.be/kIFwqp9B7pQ) ?


///----------------------PR README --------------------///


You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Comment by eshaan5** [comment]
Hey @Lakhankumawat !

I've added the changes that you mentioned. Kindly review my PR now.
Thanks

**Comment by miraj0507** [comment]
This looks good to me ready to merge @n4i9kita kindly have a look for the same

**Pull request closed without merging**